doc/introduction/interfaces.rst

```diff
@@ -3,17 +3,99 @@
 
 .. _intro_interfaces:
 
-Interfaces and training
-=======================
+Gradients and training
+======================
 
 PennyLane offers seamless integration between classical and quantum computations. Code up quantum
-circuits in PennyLane, and connect them easily to the top scientific computing and machine learning
-libraries.
+circuits in PennyLane, compute :doc:`quantum gradients <glossary/quantum_gradient>`, and
+connect them easily to the top scientific computing and machine learning libraries.
+
+Gradients
+---------
+
+When creating a QNode, you can specify the :doc:`differentiation method
+<glossary/quantum_differentiable_programming>` that PennyLane should use whenever the gradient of
+that QNode is requested.
+
+.. code-block:: python
+
+    @qml.qnode(dev, diff_method="parameter-shift")
+    def circuit(x):
+        qml.RX(x, wires=0)
+        return qml.probs(wires=0)
+
+PennyLane currently provides the following differentiation methods for QNodes, divided into
+the following three categories.
+
+.. note::
+
+    If not specified, the default differentiation method is ``diff_method="best"``. PennyLane
+    will attempt to determine the *best* differentiation method given the device and interface.
+    Typically, PennyLane will prioritize device-provided gradients, backpropagation, parameter-shift
+    rule, and finally finite-differences, in that order.
+
+Device gradients
+~~~~~~~~~~~~~~~~
+
+* ``"device"``: Queries the device directly for the gradient.
+  Only allowed on devices that provide their own gradient computation.
+
+Reverse-mode differentiation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following methods only support state-vector simulators, they are **not** hardware compatible;
+they are only supported on *statevector* simulator devices such as :class:`default.qubit
+<~.DefaultQubit>`.
+
+However, for rapid prototyping on simulators, these methods typically out-perform forward-mode
+accumulators such as the parameter-shift rule and finite-differences. For more details, see the
+:doc:`quantum backpropagation <demos/tutorial_backprop>` demonstration.
+
+* ``"backprop"``: Use standard backpropagation.
+
+  This differentiation method is only allowed on simulator
+  devices that are classically end-to-end differentiable, for example
+  :class:`default.qubit <~.DefaultQubit>`. This method does *not* work on devices
```

Comment:
Are we purposely simplifying here (i.e., hiding the fact we swap out `default.qubit` for its differentiable versions)? I think it makes sense... :thinking:

Reply:
I think that makes sense as well, since this is only a simplified overview.

Reply:
Yep, I'm purposely simplifying :) I think this is a case where providing the full details will just be more confusing.

I'm thinking we should move towards a case where `default.qubit.tf` etc. are no longer mentioned; instead, we simply refer to `default.qubit` supporting tf, autograd, jax.